File: backend/src/main/java/com/equisoft/dca/backend/location/model/LocationType.java
package com.equisoft.dca.backend.location.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum LocationType implements BaseEnum<Integer> {

	NETWORK(1, "Network"),
	FTP(2, "FTP");

	private final int id;

	private final String name;

	LocationType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: react/src/lib/domain/entities/LocationType.ts
import BaseEnum from './BaseEnum';

export default class LocationType extends BaseEnum {
    public static NETWORK = new LocationType('1', 'location:locationTypeEnum:network');
    public static FTP = new LocationType('2', 'location:locationTypeEnum:ftp');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}
